pkg/api/datasources.go

```diff
@@ -447,10 +461,25 @@ func (hs *HTTPServer) AddDataSource(c *contextmodel.ReqContext) response.Respons
 			return resp
 		}
 	}
+
 	if err := validateJSONData(c.Req.Context(), cmd.JsonData, hs.Cfg, hs.Features); err != nil {
 		return response.Error(http.StatusBadRequest, "Failed to add datasource", err)
 	}
 
+	// check if user has given LBAC rules and have permissions
+	if hs.Features.IsEnabled(featuremgmt.FlagTeamHttpHeaders) {
+		teamHTTPHeaders, err := datasources.GetTeamHTTPHeaders(cmd.JsonData)
+		if err != nil {
+			return response.Error(http.StatusInternalServerError, "Failed to add datasource", err)
+		}
+		if len(teamHTTPHeaders) > 0 {
```

Comment:
What if I removed all headers? Then `len(teamHTTPHeaders)` will be 0 and access control check is skipped. RBAC check should be performed on every `teamHttpHeaders` field update.

Reply:
i have this check for AddDataSource , as it is not changing only Adding. Since we do not have anything to compare to either I have this check in place